### PR TITLE
feat(example): survey with token actions

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change

1. show a survey with a token 
2. button to detect if the user has responded to the survey with a token

| Surveys Screen  | 
|-----------|
|Screenshot_1709047586|

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Jira ID = MOB-13920 , MOB-13921

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by mnoseir** [comment]
### Coverage Report
| Label | Coverage | Status |
| ----- | -------- | ------ |
| JavaScript | 96.3% | ✅ |
| Android | 47.8% | ✅ |
| iOS | 41.7% | ✅ |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eab1887f93edb9b17040ac7b2a635df16a2cb5f2
</p>

File: examples/default/src/screens/SurveysScreen.tsx [hunk 1]
```diff
@@ -16,6 +45,46 @@ export const SurveysScreen: React.FC = () => {
...
+
+          <Button
+            mt="4"
+            onPress={async () => {
+              if (validateUserAttributeForm()) {
+                const hasResponded = await Surveys.hasRespondedToSurvey(surveyToken);
+                toast.show({
+                  description: hasResponded ? 'YES' : 'NO',
+                });
+              }
```
Comment (abdelhamid-f-nasser) [inline examples/default/src/screens/SurveysScreen.tsx:82]: How about we refactor this to a method ?

**Review by abdelhamid-f-nasser — CHANGES_REQUESTED** [review]
@ahmedAlaaInstabug Great effort and neat UI 👏 👍 , please check the suggestion.

File: examples/default/src/screens/SurveysScreen.tsx (hunk 1 above)
Comment (ahmedAlaaInstabug) [reply examples/default/src/screens/SurveysScreen.tsx]: fixed

**Review by abdelhamid-f-nasser — APPROVED** [review]

**Pull request merged**